src/videojs-http-streaming.js

```diff
@@ -320,6 +320,13 @@ class HlsHandler extends Component {
         this.masterPlaylistController_.smoothQualityChange_();
       }
     });
+
+    this.on(this.tech_, 'seeking', function() {
```

Comment:
It may be worth us adding a comment essentially saying that the middleware can't catch looping behavior because the tech fires a seeking event and we don't get that through the middleware, thus we listen for that specific case.